### ISSUE TITLE
UIViewController not key value coding-compliant

### ISSUE THREAD

**Issue by thomas-oo** [52inc/Pulley#29]
I'm getting an exception when I try to switch to a pulleyViewController

Terminating app due to uncaught exception 'NSUnknownKeyException', reason: '[<UIViewController 0x143d485f0> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key drawerContentContainerView.

As per instructions, I added two container views in the pulleyviewcontroller and linked the outlet drawerContentContainer to a container view. And, I also linked the outlet primaryContentContainer to the other container view. These two container views segue to ordinary UIViewControllers.

I did not do any customization but I am getting this exception. My project is primilary objective-c but I don't think this has much to do with it. I even tried to use a forked project https://github.com/tadelv/Pulley/tree/objc_friendly but the exception still arises. 

TLDR: followed insturctions to setup PulleyViewController in interface builder, NSUnknownKeyException arises

**Comment by brendan09** [comment]
It looks like you haven't set the subclass of the view controller to "PulleyViewController" in your Storyboard.

If you had, this would say "PulleyViewController" instead of "UIViewController":

` '[<UIViewController 0x143d485f0> setValue:forUndefinedKey:]: `

Checkout the example project in the main repo for a working example.

**Comment by thomas-oo** [comment]
I finally got it!
I did subclass the view controller to PulleyViewController but when I checked your demo project, I noticed the module was Pulley and that I hadn't set a module in my project. **I set the module to Pulley and voila, it worked!** I guess if you don't set the module, it can't find the PulleyViewController?

Anyways I'm so happy, I've been wanting to make this work for so long! great work on this and cheers

**Comment by brendan09** [comment]
Ah! That would do it! Nice find. Let me know if you have any questions.

**Comment by trusk89** [comment]
Just realised this 3 seconds before finding this thread. Maybe this piece of info is worth to be added to the Readme?

**Comment by funkenstrahlen** [comment]
Stumbled across this problem too. Would never have found the solution without finding this.

**Comment by brendan09** [comment]
Added a note about this (with a link to the issue) to the Interface Builder section of the README. Thanks for the heads up everyone!

**Comment by alpa99** [comment]
<img width="732" alt="bildschirmfoto 2017-10-04 um 19 08 58" src="https://user-images.githubusercontent.com/29914517/31189167-8e592bd2-a937-11e7-998c-770496931d56.png">

how can i fix this ? 
Never had this problem before when i used the pulley lib for other projects with swift 3

**Comment by mickmgit** [comment]
Xcode 9.1.4, latest version of Pulley.  If I have the Module as Pulley, I get: 

Unknown class _TtC6Pulley20PulleyViewController in Interface Builder file.

If I set the Module to None, I get:

Unknown class PulleyViewController in Interface Builder file.

Both situations crash.

If I check "Inherit Module From Target", then it works.

**Comment by brendan09** [comment]
That is an expected Interface Buildee behavior, and not related to Pulley.

**Comment by EmanueleVolpato99** [comment]
I can't use the pulley library programmatically without connecting it to the storyboard, what should I do?
